### ISSUE TITLE
Support new unary operators

### ISSUE THREAD

**Issue by ngthanhtrung** [typelevel/machinist#9]
Hi,

Suppose I have following code:

```scala
import machinist.DefaultOps

trait SecurityLevel[T] {
  def literal: String
}

implicit class SecurityLevelOps[T: SecurityLevel](val lvl: T) {
  def literal: String = macro DefaultOps.unop[String]
}
```

But currently, `unop` is defined as

```scala
def unop[R](c: Context)(): c.Expr[R] = ???
```

which requires my `literal` method to have parentheses as well. It would be great if we have a unary operator without parentheses.

In addition, because `literal` method in `SecurityLevel[T]` is defined without parameters (it cares just about the type `T`, doesn't need to care about the value, i.e. it's constant on the type), one more unary operator for this use case is also needed.

Do these operators align with `machinist` roadmap? If yes, I would be happy to submit a pull request. Thanks.

**Comment by non** [comment]
@ngthanhtrung Hi there! I'm sorry I missed this issue. I do think supporting these kinds of unary operators is important. I'll see what I can do -- I seem to recall it being difficult to support parameterless def methods at some point, but maybe there's no problem here.

**Comment by non** [comment]
It looks like it's quite straightforward. I'm going to add `unop0`, keeping `unop` behaving as it does.

**Comment by non** [comment]
So `unop0` is now available in 0.5.0. Apologies again for taking so long to respond!

**Comment by ngthanhtrung** [comment]
Thanks!
